### PR TITLE
Updated Icelandic post code field

### PR DESCRIPTION
**Pull request by HenrikOssipoff**
It seems like there were missing a *lot* of Icelandic post codes. It also appears that the list contained a lot of post codes that are actually not valid post codes.

The list have been updated based on the official data bank from Postur.is: https://www.postur.is/en/about-us/post-offices/post-codes/data-files/

I'm not sure if this is actually a valid change, given that this _removes_ some post codes? Should the new ones just be added, and the old ones kept?

Also note that some cities have seemingly changed names based on the newest data.

**All Changes**

- [X] Add an entry to the docs/changelog.rst describing the change.

- [X] Add an entry for your name in the docs/authors.rst file if it's not
      already there.

### CODE REVIEW

**Comment by benkonrath** [comment]
Thanks for the PR and sorry for the delay. The next release will be 3.0 which can have breaking changes. The timing of this fix is good but we also have to consider this even if we're weren't doing a major version bump - we'd just have to include in a different way.

The PR looks good. The only addition would be specifically mentioning the breaking change as a new line under the notice for the removal of the checksum functions. Thanks.

**Comment by HenrikOssipoff** [comment]
@benkonrath Thanks for the review. I've added a notice and re-worded the change log slightly. Let me know if it's good, and I'll squash the PR.

File: docs/changelog.rst [hunk 1]
```diff
@@ -25,11 +27,14 @@ Modifications to existing flavors:
...
   `si.forms.SIEMSOField`, `si.forms.SITaxNumberField` and `za.forms.ZAIDField`
   (`gh-392 <https://github.com/django/django-localflavor/pull/392>`_).
+- Updated Icelandic `IS_POSTALCODES` with missing entries, updated spelling of entries, and removed non-existing ones.
+  (`gh-394 <https://github.com/django/django-localflavor/pull/394>`_).
 
 Other changes:
 
 - Removed deprecated `generic.checksums.luhn` and `generic.checksums.ean` functions
   (`gh-379 <https://github.com/django/django-localflavor/pull/379>`_).
+- Re
```
Comment (benkonrath) [inline docs/changelog.rst]: Was this included by mistake?

Comment (benkonrath) [reply docs/changelog.rst]: I've fixed this.

**Pull request merged**